python-checks/src/test/resources/checks/deadStore.py

```diff
@@ -220,4 +220,46 @@ def falsy_or_true_initialization():
     x4 = 42
     print(x4)
 
+def multiple_assignment():
+    a, b = foo() # Noncompliant {{Rename "a" to "_" as it is not used after assignment.}}
+    print(b)
+    a = 42
+    print(a)
+
+def foreach_block_nok(list):
+    i = 42 # Noncompliant
+    for i in list:
+        print i
+
+def with_stmt_ok():
+    x = 42
+    with open(x): # OK
+        pass
+    print(x)
+
+def with_stmt_nok():
+    x = 42  # Noncompliant
+    with open() as x:
+        pass
+    print(x)
+
+def with_stmt_ok_2():
+    with open1() as x:  # OK
+        pass
+    with open2() as x:
+        pass
+    print(x)
 
+def used_in_subfunction():
+    var = 42
+    print(var)
+    def inner(): print(var)
+    var = 43 # OK
+    inner()
```

Comment:
As discussed, we should document the fact that we might have a FN if there is a dead store after the inner function call